src/Mvc/Mvc.RazorPages/src/ApplicationModels/IPageApplicationModelPartsProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels
+{
+    public interface IPageApplicationModelPartsProvider
```

Comment:
I'm slightly partial to making this an abstract base type, to make it easier to evolve it if we ever need to, but this looks really great as is

Reply:
I agree it would allow us to add new functionality to the parts provider without breaking existing third-party parts providers. What I don't like about this approach is, that we will be extending silently parts providers, which were created by third-party. They customized it completely and then we will push new code without them knowing.
I would say, that if we want to allow partial update of the parts provider (thus allowing people to accept default functionality and change/extend only parts), we should expose DefaultPageApplicationModelPartsProvider and make methods virtual. Since this is not desired, I would keep the current solution - if somebody creates new parts provider and we add new functionality, he will get compilation error and acknowledges, that there is new stuff he has to implement.
What is your view of this?

Reply:
>  What I don't like about this approach is, that we will be extending silently parts providers, which were created by third-party.

While that's true, it may hamstrings the ability to add new feature and end up with code that ends up downcasting: 
https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/ApplicationModels/DefaultApplicationModelProvider.cs#L470-L479.

I'm ok leaving this as is now. That said, could you add a comment to the interface since it's new API:

> Provides parts that are used to construct a `<see cref="PageApplicationModel" /> instance`

Reply:
Okay - comment added; it is definitely appropriate to have the comment there - thanks!